### DISCUSSION TITLE
Ionic React project not intializing Firebase properly for iOS

### DISCUSSION THREAD

**Discussion by craigm26** [capawesome-team/capacitor-firebase#577] [Q&A]
[Link to Branch issue](https://github.com/craigm26/bike-bus/tree/craigm26/issue44)

I'll update later. kids.

**Comment by robingenz** [comment]
Please describe your issue in detail and append screenshots and logs if possible.

**Reply by craigm26** [reply]
I'll try to post logs later (from iOS simulator). For now here's where I try to log in (welcome.tsx) and where the signInWithCredential just stops the process:

Here's Welcome.tsx: 

![image](https://github.com/capawesome-team/capacitor-firebase/assets/8204366/68a724a9-ac15-43f0-b700-9e72d503f4ea)

Here's [useAuth](https://github.com/craigm26/bike-bus/blob/main/src/useAuth.ts).tsx: 

![image](https://github.com/capawesome-team/capacitor-firebase/assets/8204366/adf603e0-dbd6-4072-84ba-b7550e189512)


I think overall, I do have a different pattern for useContext for auth that I tried to implement as a result of this, but that doesn't seem to work either. 

More to come: logs and other screenshots.

**Reply by craigm26** [reply]
log result and log credential show up in the simulator and it looks valid. Afterwards I get: 

10.23.1 - [FirebaseAuth][I-AUT000018] Error getting App Check token; using placeholder token instead. Error: Error Domain=com.google.app_check_core Code=4 "The attestation provider DeviceCheckProvider is not supported on current platform and OS version." UserInfo={NSLocalizedFailureReason=The attestation provider DeviceCheckProvider is not supported on current platform and OS version.}

then 

10.23.1 - [FirebaseAuth][I-AUT000017] Has valid access token. Estimated expiration date: 2024-04-02 20:38:49 +0000, current date: 2024-04-02 19:38:49 +0000

and finally: 

10.23.1 - [FirebaseAuth][I-AUT000018] Error getting App Check token; using placeholder token instead. Error: Error Domain=com.google.app_check_core Code=4 "The attestation provider DeviceCheckProvider is not supported on current platform and OS version." UserInfo={NSLocalizedFailureReason=The attestation provider DeviceCheckProvider is not supported on current platform and OS version.}



The welcome.tsx has a button initiating the auth flow: 
```

const handleGoogleSubmit = async () => {
    try {
      console.log('starting signInWithGoogle');
      const userCredential = await signInWithGoogle();
      console.log('userCredential', userCredential);
      if (!userCredential) {
        throw new Error('userCredential is null, Failed to sign in with Google.');
      }
      if (userCredential) {
        await processUser(userCredential.user);
      }
    } catch (error) {
      console.error('Error during Google sign in:', error);
    }
  };

  const processUser = async (user: User | null | undefined) => {
    if (user) {
      await checkAndUpdateAccountModes(user.uid);
      const username = user.displayName;
      if (username) {
        setIsLoggedIn(true);
        setUsername(username);
      } else {
        setIsLoggedIn(true);
        setUsername('Anonymous Username');
      }
    }
  }; 
```

and here's where that signInWithGoogle(); goes to : [useAuth.tsx](https://github.com/craigm26/bike-bus/blob/main/src/useAuth.ts): 

```
const signInWithGoogle = async (): Promise<UserCredential | null> => {
    try {
      const provider = new GoogleAuthProvider();
      console.log('provider:', provider);
      if (Capacitor.isNativePlatform()) {
        console.log('isNativePlatform');
        console.log('FirebaseAuthentication:', FirebaseAuthentication);
        const result = await FirebaseAuthentication.signInWithGoogle();
        console.log('result:', result);
        const credential = GoogleAuthProvider.credential(result.credential?.idToken);
        console.log('credential:', credential);
        return await signInWithCredential(firebaseAuth, credential);
      } else {
        // Use Firebase JS SDK for web authentication
        return await signInWithPopup(firebaseAuth, provider);
      }
    } catch (error) {
      // Handle error
      if ( error instanceof Error ) {
        console.error('Error signing in with Google:', error);
        setError(error.message);
        throw new Error(error.message);
        return null;
      } else {
        console.error('Error signing in with Google:', error);
        setError('An unknown error occurred. Please try again later.')
        return null;
      }
      return null;
    }
  };
```

**Reply by craigm26** [reply]
added "AppCheckDebugProvider" to AppDelegate file in iOS. Returns a 403: 

App attestation failed
status: "permission denied"

**Reply by craigm26** [reply]
set debug token in Firebase and seems to function but still seems to stop after it tries to process signInWithCredential within the google login in the useAuth page. 

log results after the useCredential is logged: 
10.23.1 - [FirebaseAnalytics][I-ACS023087] User property set. Name, value: lifetime_user_engagement (_lte), 13586185
10.23.1 - [FirebaseAnalytics][I-ACS023087] User property set. Name, value: session_user_engagement (_se), 250040

**Reply by craigm26** [reply]
after the log credential is parsed and signInWithCredential stops the flow, we get this

10.23.1 - [FirebaseAuth][I-AUT000017] Has valid access token. Estimated expiration date: 2024-04-02 21:45:53 +0000, current date: 2024-04-02 20:45:53 +0000

**Comment by craigm26** [comment]
Going to simplify this issue: Try to log in anonymously. 
```
    const signInAnonymously = async (): Promise<UserCredential> => {

      const isIOS = navigator.userAgent.match(/iPhone|iPad|iPod/i);

      if (isIOS) {
        console.log('isIOS');
        try {
          console.log('signInAnonymously on iOS called');
          const authInstance = firebaseAuth;
          console.log('authInstance:', authInstance);
          const userCredential = await firebaseSignInAnonymously(authInstance);

          console.log('userCredential:', userCredential);
          const user = userCredential.user;

          if (user) {
            console.log('user exists');
            console.log('user.uid:', user.uid);
            const userRef = doc(db, 'users', user.uid);
            console.log('userRef:', userRef);
            const userData = {
              accountType: "Anonymous",
              enabledAccountModes: ["Anonymous"]
            };
            console.log('userData:', userData);

            await setDoc(userRef, userData, { merge: true });
            console.log('userRef set with userData');

            const mappedUser = await mapFirebaseUserToUserData(user);
            console.log('mappedUser:', mappedUser);
            setUser(mappedUser);
          }

          return userCredential;
        } catch (error) {
          console.error('Error signing in anonymously:', error);
          throw error;
        }
      } else {

        try {
          console.log('signInAnonymously called');
          const userCredential = await firebaseSignInAnonymously(firebaseAuth);
          console.log('userCredential:', userCredential);
          const user = userCredential.user;

          if (user) {
            console.log('user exists');
            console.log('user.uid:', user.uid);
            const userRef = doc(db, 'users', user.uid);
            console.log('userRef:', userRef);
            const userData = {
              accountType: "Anonymous",
              enabledAccountModes: ["Anonymous"]
            };
            console.log('userData:', userData);

            await setDoc(userRef, userData, { merge: true });
            console.log('userRef set with userData');

            const mappedUser = await mapFirebaseUserToUserData(user);
            console.log('mappedUser:', mappedUser);
            setUser(mappedUser);
          }

          return userCredential;
        } catch (error) {
          console.error('Error signing in anonymously:', error);
          throw error;
        }
      }
    };
```
This results in 
⚡️  [log] - isIOS
⚡️  [log] - signInAnonymously on iOS called
⚡️  [log] - authInstance: --- valid apiKEY and authDomain
